sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/BasicWriteStatsTracker.scala

```diff
@@ -57,7 +60,14 @@ class BasicWriteTaskStatsTracker(hadoopConf: Configuration)
   private def getFileSize(filePath: String): Long = {
     val path = new Path(filePath)
     val fs = path.getFileSystem(hadoopConf)
-    fs.getFileStatus(path).getLen()
+    try {
+      fs.getFileStatus(path).getLen()
+    } catch {
+      case e: FileNotFoundException =>
+        // may arise against eventually consistent object stores
+        logInfo(s"File $path is not yet visible", e)
```

Comment:
Could you update the log message and indicate the size zero might be wrong? For example negative caching in S3

Reply:
say "Reported file size in job summary may be invalid"?